core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -526,6 +537,19 @@ private[spark] class ExecutorAllocationManager(
       allocationManager.synchronized {
         stageIdToNumTasks(stageId) = numTasks
         allocationManager.onSchedulerBacklogged()
+
+        var numTasksPending = 0
```

Comment:
Ah just looked deeper into the code and it seems like localities are often empty.  So nothing to be done here, sorry for the noise.